Mage.Sets/src/mage/cards/w/WormfangCrab.java

```diff
@@ -0,0 +1,105 @@
+import mage.util.CardUtil;
+
+/**
+ * @author xenohedron
+ */
+
+public final class WormfangCrab extends CardImpl {
+
+    public WormfangCrab(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{U}");
+        
+        this.subtype.add(SubType.NIGHTMARE);
+        this.subtype.add(SubType.CRAB);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(6);
+
+        // Wormfang Crab is unblockable.
+        this.addAbility(new CantBeBlockedSourceAbility());
+
+        // When Wormfang Crab enters the battlefield, an opponent chooses a permanent you control other than Wormfang Crab and exiles it.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new WormfangCrabExileEffect(), false));
+
+        // When Wormfang Crab leaves the battlefield, return the exiled card to the battlefield under its owner's control.
+        this.addAbility(new LeavesBattlefieldTriggeredAbility(new ReturnFromExileForSourceEffect(Zone.BATTLEFIELD), false));
+    }
+
+    private WormfangCrab(final WormfangCrab card) {
+        super(card);
+    }
+
+    @Override
+    public WormfangCrab copy() {
+        return new WormfangCrab(this);
+    }
+}
+
+class WormfangCrabExileEffect extends OneShotEffect {
+
+    public WormfangCrabExileEffect() {
+        super(Outcome.Exile);
+        this.staticText = "an opponent chooses a permanent you control other than {this} and exiles it";
+    }
+
+    public WormfangCrabExileEffect(final WormfangCrabExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WormfangCrabExileEffect copy() {
+        return new WormfangCrabExileEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        MageObject sourceObject = source.getSourceObject(game);
+        if (controller == null || sourceObject == null) {
+            return false;
+        }
+        TargetOpponent targetOpponent = new TargetOpponent(true);
+        controller.choose(outcome, targetOpponent, source, game);
+        Player opponent = game.getPlayer(targetOpponent.getFirstTarget());
+        if (opponent == null) {
+            return false;
+        }
+        game.informPlayers(sourceObject.getLogName() + ": "
+                + controller.getLogName() + " has selected " + opponent.getLogName() + " to choose a permanent to exile");
+        FilterPermanent filter = new FilterPermanent();
+        filter.add(AnotherPredicate.instance);
+        filter.add(new ControllerIdPredicate(controller.getId()));
+        TargetPermanent target = new TargetPermanent(filter);
+        target.setNotTarget(true);
+        target.setTargetName("a permanent that player controls");
+        if (!opponent.choose(outcome, target, source, game)) {
+            return false;
+        }
+        Card cardToMove = game.getCard(target.getFirstTarget());
```

Comment:
you need to use `getPermanent` here instead of `getCard`

Reply:
thanks, that is what I was missing